test/integration/functions/op.equality.test.ts

```diff
@@ -58,10 +58,16 @@ describe('evaluation of \'=\'', () => {
     runTestTable({
       ...config,
       testTable: `
+        "test1"@en "test1"@en = true
+        "test1" "test1" = true
         empty empty = true
         empty aaa   = false
         aaa   aaa   = true
         aaa   bbb   = false
+        "test1" "test2" = false
```

Comment:
Is it normal that the case without language strings return `false` and with language string fails? Both are literals right? I'm a little confused here. 

Additionally, the original [Communica issue](https://github.com/comunica/comunica/issues/1146) states:
> Any FILTER on ?o in the example fails (with or without language string, = or !=).

So there it seems to also fail without language strings?  (This issue might refer to `"test" = "test"@en`, if this is the case, a test regarding this case is in place.)

Reply:
## The current situation with Sparqlee:

Treats as Literal in the case of equality checks:
- "test"
- "test"@en

Treats as strings and checks them here: https://github.com/comunica/sparqlee/blob/master/lib/functions/RegularFunctions.ts#L96

- test

## What should we do? 

I think the consensus in other issues is that Literals and strings inside Sparql should be treated as Literals. 

See:

- https://github.com/comunica/sparqlee/issues/2
- https://github.com/rdfjs/data-model-spec/issues/127#issuecomment-385320677
- 
Especially the last part of the last link seems important:

> In 2014, the RDF 1.1 WG decided that the pain of maintaining a distinction between simple literal and xsd:string-typed literal exceeded the pain of migration to a model where everything had a datatype. In the latter model, parsers that appeared to have a simple literal form were expected to produce literals with type xsd:string. You are experiencing some of that migration pain. Apologies for not getting this right in 1999.

That would mean we have to:

- Treat `test` and `"test"@en` the same AND both should throw errors in case of a comparison with an other string that is not equal.

This mean changing internals of Sparqlee to pick up 'plain strings' as Literals.

@jitsedesmet What are your thoughts?

Reply:
Yes, I think this means we should just handle simple literals as xsd:string-typed literals as soon as they enter sparqlee. I'll ask for @rubenworks his opinion. If he gives his thumbs up, I suggest you look at the files in the transformers directory.

Reply:
> Is it normal that the case without language strings return false and with language string fails? Both are literals right? I'm a little confused here.

Note that the test here checks equality for `"test1"` and `"test2"`, which are different strings, hence the failure.

> I think this means we should just handle simple literals as xsd:string-typed literals as soon as they enter sparqlee

Indeed. AFAIK, this is how things work today.
Even the RDF/JS data model has been designed like this. An untyped literal string will default to the string datatype.

> Treat test and "test"@en the same AND both should throw errors in case of a comparison with an other string that is not equal.

No, `"test"` and `"test"@en` are not the same. The one has a string datatype without language, and the other is a language-tagged-string with the `en` language.

Reply:
> No, "test" and "test"@en are not the same. The one has a string datatype without language, and the other is a language-tagged-string with the en language.

Sure, I agree, I meant something different with my comment. I meant:

Treat all the following cases 'the same', the same equality check:

_Should pass:_
test1 = test1
"test1"@en = "test1"@en

_Should not pass:_
test1 = test2
"test1"@en = "test2"@en
"test1"@en = "test1"
"test1"@en = test1

The point being that they all undergo the same equality check, 
which at the moment in Sparqlee is not the case. 

I tested the code and plain string equality seemed to happen here: https://github.com/comunica/sparqlee/blob/master/lib/functions/RegularFunctions.ts#L96

Term equality went through this: https://github.com/comunica/sparqlee/blob/master/lib/functions/RegularFunctions.ts#L106

Reply:
> Treat all the following cases 'the same', the same equality check:

Yep, that's correct, according to the [spec](https://www.w3.org/TR/rdf-concepts/#section-Literal-Equality).

> The point being that they all undergo the same equality check,
which at the moment in Sparqlee is not the case.

I suspect the datatype and language checks would be part of [this line](https://github.com/comunica/sparqlee/blob/master/lib/functions/RegularFunctions.ts#L109).

Reply:
> Yes, I think this means we should just handle simple literals as xsd:string-typed literals as soon as they enter sparqlee. I'll ask for @rubenworks his opinion. If he gives his thumbs up, I suggest you look at the files in the transformers directory.

Shall I proceed?

Reply:
Yes.
However, the approach I suggested regarding `TermTransfromer` might not be correct. 
I'll try to summarize and reason how we can proceed.

The spec tells us `"test"` should become `"test"^^xsd:string` as soon as it is parsed.
Similarly, `"test"@en` should become `"test"@en^^rdf:langString` or `"test"@en^^xsd:string`.
(The docs should be read for this).
I think it'll be `rdf:langString` because of https://www.w3.org/TR/rdf11-concepts/#h3_section-Graph-Literal, but it should be checked.
I think @rubensworks had found an official document regarding the transformation of plain literals.
I don't know if this is handled by sparqlee or not, and I don't know whether we use a package for this.

That being said, we should look what behavior the spec of equality describes.
Equality is described in: https://www.w3.org/TR/sparql11-query/#func-isLiteral#OperatorMapping 
For strings this is: https://www.w3.org/TR/xpath-functions/#func-compare
For language string I have not found a custom equality function, meaning it should be checked with the term equal: https://www.w3.org/TR/sparql11-query/#func-RDFterm-equal

If all of this is right, I think the implementation in the current PR is right. (?) I will however first check if I find anything for the language equality.

Reply:
Alright. Since I don't find any special treatment for languageStrings, this implementation should be correct and ready to merge.